crates/agglayer-storage/src/tests/db_generator.rs

```diff
@@ -0,0 +1,773 @@
+    let state_path = base_path.join("state");
+    let state_result = generate_state_db(&state_path, config)?;
+    result
+        .entries_per_cf
+        .extend(state_result.entries_per_cf.clone());
+    result.network_ids = state_result.network_ids.clone();
+    result.certificate_ids = state_result.certificate_ids.clone();
+    result.certificates = state_result.certificates.clone();
+
+    // Generate pending database
+    let pending_path = base_path.join("pending");
+    let pending_result = generate_pending_db(&pending_path, config, &state_result)?;
+    result.entries_per_cf.extend(pending_result.entries_per_cf);
+
+    // Generate epochs database
+    let epochs_path = base_path.join("epochs");
+    let epochs_result = generate_epochs_db(&epochs_path, config, &state_result)?;
+    result.entries_per_cf.extend(epochs_result.entries_per_cf);
+
+    // Generate debug database
+    let debug_path = base_path.join("debug");
+    let debug_result = generate_debug_db(&debug_path, config, &state_result)?;
+    result.entries_per_cf.extend(debug_result.entries_per_cf);
+
+    Ok(result)
+}
+
+/// Generate state database with all column families
+pub fn generate_state_db(
+    path: &Path,
+    config: &GeneratorConfig,
+) -> Result<GenerationResult, crate::storage::DBError> {
+    let db = DB::open_cf(path, state_db_cf_definitions())?;
+    let mut rng = rand::rngs::StdRng::seed_from_u64(config.seed);
+
+    let mut result = GenerationResult {
+        entries_per_cf: std::collections::HashMap::new(),
+        network_ids: Vec::new(),
+        certificate_ids: Vec::new(),
+        certificates: std::collections::HashMap::new(),
+    };
+
+    // Generate data for each network
+    for network_idx in 0..config.num_networks {
+        let network_id = NetworkId::new(network_idx + 1);
+        result.network_ids.push(network_id);
+
+        // Initialize the first local exit root for this network
+        let initial_ler: LocalExitRoot = LocalExitTree::<32>::default().get_root().into();
+        let mut prev_local_exit_root = initial_ler;
+
+        // Generate certificates for this network
+        for height in 0..config.certificates_per_network {
+            let height = Height::from(height);
+
+            // Vary parameters based on height to create diverse test data
+            let num_bridge_exits = (height.as_u64() % 5) as usize; // 0-4 bridge exits
+            let aggchain_data_type = match height.as_u64() % 4 {
+                0 => AggchainDataType::Ecdsa,
+                1 => AggchainDataType::Generic,
+                2 => AggchainDataType::MultisigOnly { num_signers: 3 },
+                _ => AggchainDataType::MultisigAndAggchainProof { num_signers: 5 },
+            };
+            let version = match height.as_u64() % 3 {
+                0 => SignatureCommitmentVersion::V2,
+                1 => SignatureCommitmentVersion::V3,
+                _ => SignatureCommitmentVersion::V5,
+            };
```

Comment:
Nit: Why this height based cyclic approach rather than using rng? The drawback of the current approach is that there will not be good coverage if things change somewhat. For example, if commitment version V6 is introduced, then Ecdsa will always be paired with V2, Generic with V3, Multisig with V5 and MultisigWithProof with V6, and then back to beginning.

Reply:
Wanted to have repeatable data generation - so that we can regenerate the same data every time. But yeah, with rng with seed it is also possible.